.github/workflows/Produce_build_and_test_on_main.yml

```diff
@@ -0,0 +1,18 @@
+name: Produce build main
+
+on:
+  push:
+    branches:    
+      - master
```

Comment:
I think this should be main, not master

Reply:
There was conversation about main -> master in Teams.

Reply:
Main branch is `main` in this repository, not `master`.